libreyolo/models/rfdetr/transformer.py

```
@@ -500,21 +595,101 @@ def forward(self, srcs, masks, pos_embeds, refpoint_embed, query_feat):
+                refpoint_embed = torch.cat([torch.cat([chunk, reg_ref], dim=1) for chunk in ref_chunks], dim=1)
+                if init_kp_ref_xy is not None:
+                    num_keypoints = init_kp_ref_xy.shape[2]
+                    reg_kp_xy = self.register_ref_points[:, :2].sigmoid()
+                    reg_kp_xy = reg_kp_xy.unsqueeze(0).unsqueeze(2).expand(bs, -1, num_keypoints, -1)
+                    kp_ref_chunks = list(init_kp_ref_xy.split(original_num_queries_per_group, dim=1))
+                    init_kp_ref_xy = torch.cat([torch.cat([chunk, reg_kp_xy], dim=1) for chunk in kp_ref_chunks], dim=1)
+
+            tgt_keypoints = None
+            if self.use_grouppose_keypoints:
+                if not hasattr(self, "keypoint_query_initializer"):
+                    raise ValueError("use_grouppose_keypoints=True requires keypoint initializers")
```

Ruff (0.15.17)

[warning] Avoid specifying long messages outside the exception class

(TRY003)